runtime/wasm/src/to_from/external.rs

```diff
@@ -290,8 +290,45 @@ impl ToAscObj<AscEthereumTransaction> for EthereumTransactionData {
     }
 }
 
-impl ToAscObj<AscEthereumEvent> for EthereumEventData {
-    fn to_asc_obj<H: AscHeap>(&self, heap: &mut H) -> AscEthereumEvent {
+impl ToAscObj<AscEthereumTransactionV2> for EthereumTransactionData {
+    fn to_asc_obj<H: AscHeap>(&self, heap: &mut H) -> AscEthereumTransactionV2 {
+        AscEthereumTransactionV2 {
+            hash: heap.asc_new(&self.hash),
+            index: heap.asc_new(&BigInt::from(self.index)),
+            from: heap.asc_new(&self.from),
+            to: self
+                .to
+                .map(|to| heap.asc_new(&to))
+                .unwrap_or_else(|| AscPtr::null()),
+            value: heap.asc_new(&BigInt::from_unsigned_u256(&self.value)),
+            gas_used: heap.asc_new(&BigInt::from_unsigned_u256(&self.gas_used)),
+            gas_price: heap.asc_new(&BigInt::from_unsigned_u256(&self.gas_price)),
+            input: heap.asc_new(&*self.input.0),
+        }
+    }
+}
+
+impl ToAscObj<AscEthereumEvent<AscEthereumTransactionV1>> for EthereumEventData {
```

Comment:
I'd think that we can merge this impl and the one below into one that's generic over the transaction type version.